documentation/smg/08_setting_up_jenkins_ci.md

```diff
@@ -0,0 +1,126 @@
+
+- Jenkins loads and runs the [`Jenkinsfile`](./../../tools/build_config/Jenkinsfile)
+located in `tools/build_config`.
+- The `Jenkinsfile` script runs our build script (one for Windows, one for Unix)
+and updates GitHub about the status of the build.
+
+## More Detailed Set Up
+
+### GitHub
+
+The only job required on GitHub is to set up webhooks to notify Jenkins of pull
+requests. These can be created by GitHub repository admins by opening the
+settings tab in the [main repo](https://github.com/pace-neutrons/Horace).
+
+<img src="./images/08_github_settings.png">
+
+Then selecting the `Webhooks` menu item on the left-hand side.
+
+Your webhook should have payload URL
+`https://anvil.softeng-support.ac.uk/jenkins/generic-webhook-trigger/invoke?token=<my_secret_token>`
+, content type `application/json` and only trigger on pull request events. This
+will send a json containing information about the pull request (including pull
+request number and the action taken in the pull request event) to Jenkins. Pull
+request events trigger when a pull request is:
+
+> *opened, closed, reopened, edited, assigned, unassigned, review requested,
+> review request removed, labeled, unlabeled, synchronized, ready for review,
+> locked, or unlocked.*
+
+This means that, from the webhook's JSON, you must decide if you want to trigger
+a build. usually you'd only want to trigger builds when a pull request is
+`opened` or `synchronized`.
+
+Form this page you can also re-deliver payloads that may not have reached
+Jenkins e.g. if the Jenkins servers were down.
+
+Using webhooks is a more efficient method to automatically trigger builds than
+polling, as it does not require Jenkins to query GitHub on regular intervals.
+
+### Jenkins GUI
+
+#### Required Plugins
+
+- [Jenkins Git](https://plugins.jenkins.io/git) - Clone Git repo
+- [Generic Webhook Trigger](https://plugins.jenkins.io/generic-webhook-trigger) -
+Allows GitHub to trigger build
+- [xUnit](https://plugins.jenkins.io/xunit) - Parse and display test results
+
+Follow these steps to set up the pipeline:
+
+- Create a new "Pipeline" job.
+- Enter the GitHub project URL (this creates a link to the GitHub from the pipeline).
+- Select the `This project is parameterised` option:
+    - Create the following string parameters:
+        - `AGENT`: The label of the agent to run the job on
+        - `CMAKE_VERSION`: The version of CMake to load
+        - `MATLAB_VERSION`: The (release) version of Matlab to load
+        - `GCC_VERSION`: The version of GCC to use (Linux only)
+
+  For pull requests:
+    - Select the `Generic Webhook Trigger` option and retrieve the json values:
+        - `action`: The type of pull request event this is
+        - `pull_request.number`: The pull request number on GitHub
+        - `pull_request.statuses_url`: The url to send build statuses to
+        - `pull_request.head.sha`: The HEAD sha of the pull request branch
+
+      <img src="./images/08_commit_sha.png" width=70%>
```

Comment:
this image is being horribly resized. I'd be tempted to leave all images at 100% unless you really do need it as a thumbnail of a much bigger images.